### ISSUE TITLE
Code seems confused about architecture

### ISSUE THREAD

**Issue by atz** [sul-dlss/dor-workflow-service#14]
The code has a class that exposes methods like `Dor::WorkflowService.archive_workflow()`, but then those methods require certain `@@clas_vars` to be initialized via a `Dor::WorkflowService.configure` method.  But each place these top level vars are referenced still has to check "did you call configure?"  It seems like what we really want is a regular object that is initialized by constructor every time, so that we only have to check once.  And then if you happen to have two different pieces of code touch Dor::WorkflowService, they don't collide with competing global configurations.  

Is there any reason for the way it is?  Anybody want to vouch for it?

**Comment by drh-stanford** [comment]
Everything is a class method now. I don't see why it couldn't be a singleton pattern that initialized the config stuff on first call.

**Comment by atz** [comment]
It certainly *could* employ that pattern, but I don't see any reason it *should*.  I see several reasons it shouldn't.  The `@@module_vars` are mimicking attributes.  That seems like bad behavior.  Modules are not intended to have internal states.  Modules are typically expected to be included or extended, which we can't do here without abandoning the singleton-ness of the `@@vars`.  The combination of methods and states is the proper domain of a class.  

Basically I feel like this was copied out of dor-services and overly influenced by the "everything is a module" approach used there.  Maybe I will have time to refactor it for a major version bump, but I wonder what else is using this besides Argo and Hydrus.

**Comment by atz** [comment]
@cbeer reports sdr-preservation-core and EEMS are consumers.

**Comment by atz** [comment]
Scrapped configure() for regular class initialize() in ongoing work in `2.x` branch.
